packages/utils-query-operation/lib/MaterializeBindings.ts

```diff
@@ -106,9 +106,29 @@ export function materializeOperation(
           };
         }
       }
+      let recursionResult: Algebra.Operation = materializeOperation(
+        op.input,
+        bindings,
+        algebraFactory,
+        bindingsFactory,
+        options,
+      );
+
+      // Join op.input with a values clause containing the variables that are in InitialBindings and
+      // are used in the current extend operation.
+      const inScopeVariables = [ op.variable ];
+      if (op.expression.expressionType === Algebra.expressionTypes.TERM && op.expression.term.termType === 'Variable') {
```

Comment:
This may not work correctly if you have more complicated BIND clauses, such as `BIND (($this + $this) AS $this)`.
Something like `Util.inScopeVariables` (or a variant) from sparqlalgebrajs may help.

Reply:
Would be good to also have a test case for this.

Reply:
I will look into it!